src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -105,12 +102,14 @@ bool DirectoryService::RunConsensusOnDSBlockWhenDSPrimary()
     m_consensusBlockHash.resize(BLOCK_HASH_SIZE);
     fill(m_consensusBlockHash.begin(), m_consensusBlockHash.end(), 0x77);
 
-    // kill first ds leader
-    // if (m_consensusMyID == 0 && temp_todie)
-    // {
-    //    LOG_GENERAL(INFO, "I am killing myself to test view change");
-    //    throw exception();
-    // }
+    // kill first ds leader (used for view change testing)
```

Comment:
Why need to preserve this?

Reply:
Although ugly, we got to preserve it for future view change testing. I can't think of a simple unit test to remove this part.